sdk/core/azure-common/azure/common/client_factory.py

```diff
@@ -152,6 +160,12 @@ def get_client_from_json_dict(client_class, config_dict, **kwargs):
     :param dict config_dict: A config dict.
     :return: An instantiated client
     """
+    if _is_autorest_v3(client_class):
+        raise ValueError(
+            "Auth file or JSON dict are deprecated auth approach and are not supported anymore. "
```

Comment:
This implies that these approaches are generally deprecated -- should the deprecation be specific to track 2 clients? e.g. by saying something like "azure-common factory methods are deprecated for `client_class.__name__` and are not supported anymore."

Reply:
It's under a "track2" `if` statement, is that not enough you think?

Reply:
I think it should cover the cases where this should be raised, but I just wonder if users would be led to think that these factory methods wouldn't work for non-deprecated track 1 clients as well. Does that make sense?

Reply:
This is assuming that most users wouldn't look at the source code to notice that the error is specific to this condition

Reply:
Well, I think we don't love auth file, and therefore being sligtly wide in the message is ok to me. But this is a good point to think.